packages/lisk-p2p/src/p2p_types.ts

```diff
@@ -113,6 +114,18 @@ export type P2PPeerSelectionForConnection = (
 	nodeInfo?: P2PNodeInfo,
 ) => ReadonlyArray<P2PDiscoveredPeerInfo>;
 
+export interface P2PPeerCheckCompatibility {
+	readonly networkCompatible?: P2PPeerNetworkCompatibility;
+	readonly protocolVersionCompatible?: P2PPeerProtocolVersionCompatibility;
+	readonly versionCompatible?: P2PPeerVersionCompatibility;
+}
+
+export type P2PPeerNetworkCompatibility = (nethash: string) => boolean;
```

Comment:
instead of specifying one by one, maybe
`P2PPeerNetworkCompatibility = (header: QueryObject) => boolean`
and i think we can provide the default by
checking nethash, version, and protocol version, and override them if it's provided?

Reply:
Although, I have added this check now. But by default, all the nodes need to be compatible with each other in terms of minVersion, version and nethash which should be handled in module level in IMO. Now anyone using p2p library needs to pass these values. @jondubois What do you think?

Reply:
I think it would be simpler if there was a single compatibility function instead of three different ones. The compatibility function could either throw or return a custom error on failure which can be handled inside the P2P class; then the `error.message` from the function could be used as the disconnect reason; that way the user can make the error reason be anything; so it gives them more flexibility.

Also, I think that instead of `QueryObject`, we could go one step further and just pass `incomingPeerInfo` to the function (`P2PDiscoveredPeerInfo`).

Reply:
After the discussion with @jondubois, we decided to keep `networkCompatibility` and `protocolVersionCompatibility` checks so that we avoid peers from other networks to connect by default. Also, we are going to have one function for all the compatibility checks so that user is not confined to only these 3 checks but also could provide its own check.